ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/jdbc/JDBCAbstractSplitFilterRule.java

```diff
@@ -172,13 +172,14 @@ public boolean matches(RelOptRuleCall call) {
       final HiveJdbcConverter conv = call.rel(2);
 
       RexNode joinCond = join.getCondition();
+      SqlDialect dialect = conv.getJdbcDialect();
 
-      return super.matches(call) && JDBCRexCallValidator.isValidJdbcOperation(joinCond, conv.getJdbcDialect());
+      return super.matches(call, dialect) && JDBCRexCallValidator.isValidJdbcOperation(joinCond, dialect);
```

Comment:
I think the original author meant to call `super.matches(call, dialect)` and mistakenly called `super.matches(call)`. The signature of `matches(call, dialect)` is source of confusion so to avoid similar problems in the future I would suggest removing entirely this method and call directly `canSplitFilter`. 

Moreover, it seems that `canSplitFilter` already calls `JDBCRexCallValidator.isValidJdbcOperation` internally so possibly we can remove this additional call from here.

Reply:
I've seen the change https://github.com/apache/hive/pull/2504/commits/058a0796e9ac0d0e664a8ede553d5bdb5be8ec0c. Actually I would like the whole method to go away and not only the method call. I will apply this minor change locally before merging this to master.